File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/presenter/scrap/BarCodeScrapPresenter.java
package com.c_store.zhiyazhang.cstoremanagement.presenter.scrap;

import android.os.Handler;

import com.c_store.zhiyazhang.cstoremanagement.bean.ScrapContractBean;
import com.c_store.zhiyazhang.cstoremanagement.bean.UserBean;
import com.c_store.zhiyazhang.cstoremanagement.model.MyListener;
import com.c_store.zhiyazhang.cstoremanagement.model.scrap.BarCodeScrapInterface;
import com.c_store.zhiyazhang.cstoremanagement.model.scrap.BarCodeScrapModel;
import com.c_store.zhiyazhang.cstoremanagement.view.interfaceview.BarCodeScrapView;

/**
 * Created by zhiya.zhang
 * on 2017/5/24 16:25.
 */

public class BarCodeScrapPresenter {
    private BarCodeScrapInterface anInterface;
    private BarCodeScrapView cView;
    Handler mHandler = new Handler();

    public BarCodeScrapPresenter(BarCodeScrapView view) {
        this.cView = view;
        this.anInterface = new BarCodeScrapModel();
    }

    public void getScrap(String barcode) {
        cView.showLoading();
        anInterface.getScrap(UserBean.getUser(), barcode, new MyListener() {
            @Override
            public void contractSuccess() {
            }

            @Override
            public void contractSuccess(Object object) {
                cView.hideLoading();
                cView.showView((ScrapContractBean) object);
            }

            @Override
            public void contractFailed(String errorMessage) {
                cView.hideLoading();
                cView.showFailedError(errorMessage);
            }
        });
    }

}

File: CStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/presenter/scrap/BarCodeScrapPresenter.kt
/*
package com.cstore.zhiyazhang.cstoremanagement.presenter.scrap

import com.cstore.zhiyazhang.cstoremanagement.R
import com.cstore.zhiyazhang.cstoremanagement.bean.User
import com.cstore.zhiyazhang.cstoremanagement.model.MyListener
import com.cstore.zhiyazhang.cstoremanagement.model.scrap.BarCodeScrapModel
import com.cstore.zhiyazhang.cstoremanagement.utils.ConnectionDetector
import com.cstore.zhiyazhang.cstoremanagement.view.interfaceview.GenericView
import com.zhiyazhang.mykotlinapplication.utils.MyApplication

*/
/**
 * Created by zhiya.zhang
 * on 2017/8/11 17:36.
 *//*

class BarCodeScrapPresenter(val gView:GenericView){

    val anInterface=BarCodeScrapModel()

    fun getScrap(barCode:String){
        gView.showLoading()
        if (!ConnectionDetector.getConnectionDetector().isOnline) {
            gView.showPrompt(MyApplication.instance().applicationContext.getString(R.string.noInternet))
            gView.hideLoading()
            return
        }
        anInterface.getScrap(User.getUser(),barCode, object : MyListener {
            override fun listenerSuccess() {}

            override fun listenerSuccess(`object`: Any) {
                gView.hideLoading()
                gView.requestSuccess(`object`)
            }

            override fun listenerFailed(errorMessage: String) {
                gView.showPrompt(errorMessage)
                gView.hideLoading()
            }
        })
    }
}*/
